### PR TITLE
feat(deps): update deps, support of eslint 8

### PR DESCRIPTION
**Pull request by Delagen**
Not sure if it correct, but who doesn't make errors?

### CODE REVIEW

**Comment by d-arrington** [comment]
For what it's worth, I pulled your branch into my current project and it seems to be working as intended.

**Comment by crystalfp** [comment]
Please merge! It is the last eslint plugin that does not work under eslint 8. Thanks!

**Comment by jdom** [comment]
+1 I'm currently blocked from updating `eslint` due to this plugin, which blocks us from updating `@typescript-eslint` packages, which blocks us from updating `typescript` itself now (as TypeScript 4.4 isn't supported by older eslint packages)!

**Comment by bfaulk96** [comment]
Kind of sounds like it might be time for us to ask npm to declare it up for adoption and fork it? Since none of the contributors seem to be active or responsive 🙁

**Comment by jleider** [comment]
While we are waiting for these things to be merged I created a quick and dirty release with this change and #33. 

Just add the following to your `package.json`:

`"eslint-plugin-deprecation": "https://github.com/jleider/eslint-plugin-deprecation#v1.3.0"`

**Comment by Delagen** [comment]
Published as @delagen/eslint-plugin-deprecation, also included #35

**Comment by lachieh** [comment]
Rather than continuing to fork and publish new versions, perhaps the typescript-eslint team would be interested in bringing in this rule.

@JamesHenry is there a process for this sort of thing? Is that something the typescript-eslint team would do?

Thread TL;DR:
This rule doesn't yet support Eslint 8 and the maintainer @gund has been radio silent for the last 6 months.

**Comment by jdom** [comment]
I updated my previous comment, looks like I'm no longer blocked.
>  I misjudged, I was able to update the `@typescript-eslint` packages to version 5.2, and that didn't require updating `eslint` to version 8 (I was at least able to update to 7.32, not sure if that's required).

Having said that disclaimer, I'd still like the package to support to support eslint 8, so that we can move forward. Alternatively, it would be awesome to have this rule officially supported by the TypeScript team as @lachieh suggested

**Comment by crystalfp** [comment]
Thanks a lot @Delagen for your work!
I need just an update on how to use the updated plugin.

1) Installed with `npm i -D @delagen/eslint-plugin-deprecation`
2) In eslint.yaml added `- "@delagen/eslint-plugin-deprecation"` under `plugins`
3) Under  `rules:`  I tried:
`- "@delagen/eslint-plugin-deprecation: warn"`
`- "@delagen/eslint-plugin-deprecation/deprecation: warn"`
`- deprecation/deprecation: warn (as was before)`

But always eslint responds with ` Definition for rule 'deprecation' was not found (deprecation)`

Could you help me?

**Comment by Delagen** [comment]
@crystalfp just installed it, try to investigate problem

**Comment by Delagen** [comment]
@crystalfp Seems working
```
plugins:[
"@delagen/deprecation"
],
rules:{
"@delagen/deprecation/deprecation": "warn"
}
```
What version of ESLINT are you using?

**Comment by crystalfp** [comment]
With your suggestion it works! Now I have to think of a deprecation just to test.
I'm using eslint 8.0.0
Thanks!
mario

File: package.json [hunk 1]
```diff
@@ -1,12 +1,18 @@
 {
-  "name": "eslint-plugin-deprecation",
-  "version": "0.0.0-development",
+  "name": "@delagen/eslint-plugin-deprecation",
```
Comment (timgates42) [inline package.json:2]: Looks like some of the changes include in this file might not have been intended for the upstream update?

Comment (Delagen) [reply package.json:2]: This is because of fork. I updated this project in master branch

Comment (Delagen) [reply package.json:2]: @timgates42 if you has permission to merge and publish npm, I can rebase this Mr

**Comment by lachieh** [comment]
It looks like there has been [an issue](https://github.com/typescript-eslint/typescript-eslint/issues/2620) open for a while to add a deprecation rule into the typescript-eslint project.

That same issue references another issue that talks about one of three options:

> If someone wants to port the rule to an independent eslint plugin, feel free!
> If someone wants to simply document how to use the sonar rule, they can do that instead.
> If someone wants to independently implement the rule based on [tslint's implementation](https://github.com/palantir/tslint/blob/master/src/rules/deprecationRule.ts), they can do that too.
>
> _Originally posted by @bradzacher in https://github.com/typescript-eslint/typescript-eslint/issues/1223#issuecomment-586665296_

It looks like @gund went down the first route, but given the lack of activity, it might be about time to take a look at implementing the third (implementing a rule based on the tslint implemention).

I'll likely have a look at this path over the weekend, but if anyone else wants to take a look, feel free.

File: package.json (hunk 1 above)
Comment (Delagen) [reply package.json:2]: truncated into #39

**Comment by bfaulk96** [comment]
@lachieh It's worth noting that while I fully agree this would be great to have in typescript-eslint, @Delagen forked this plugin and fixed the problems (for option 1), but the issue now is that this plugin is taking up dead space in npm.

File: tsconfig.json [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+  "compilerOptions": {
+    "noEmit": true,
+    "rootDir": ".",
+    "strict": true,
+    "declaration": true,
+    "noEmitHelpers": true,
+    "importHelpers": true,
+    "sourceMap": true,
+    "module": "CommonJS",
+    "target": "ES2015",
```
Comment (snebjorn) [inline tsconfig.json:12]: Not sure if relevant but the change from ES5 to ES2015 is a breaking change :)

ES5 is not ES2015. ES2015 is ES6.

Comment (Delagen) [reply tsconfig.json:12]: If you see supported versions of node.js by eslint, I think it can be even higher

Comment (snebjorn) [reply tsconfig.json:12]: You're right. I just saw the target change and figured I'd point it out :)

But given that ESLint 8 only supports `"node": "^12.22.0 || ^14.17.0 || >=16.0.0"` then I guess it's safe to use ES2019
https://node.green/#ES2019

**Comment by gund** [comment]
Hey folks, this has been resolved in #39!

**Pull request closed without merging**